src/components/common/Input/MinMaxRangeInput/MinMaxRangeInput.jsx

```diff
@@ -285,208 +240,182 @@ class MinMaxRangeInput extends Component {
             type="number"
             value={maxValue}
             label={' '}
-            onBlur={this.onMaxTextBlur}
-            onEnter={this.onMaxTextBlur}
-            onChange={this.onMaxTextInputChanged}
+            onBlur={onMaxTextBlur}
+            onEnter={onMaxTextBlur}
+            onChange={onMaxTextInputChanged}
             autoFocus={!focusLeftTextInput}
+            min={currentMin}
+            max={currentMax}
+            placeholder={placeholder}
           />
-          {showTooltip && (
+          {valueIsBad && (
             <Tooltip
-              className={`${inputMinClassName} ${inputMaxClassName}`}
+              className={styling}
               fixed
-              style={{ ...this.textTooltipPosition() }}
+              style={{ ...textTooltipPosition() }}
               placement="right"
             >
-              {tootipContent}
+              {valueIsOutsideRange && (
+                <p>
+                  {`Value is outside the valid range: `}
+                  <b>{`[${min}, ${max}].`}</b>
+                </p>
+              )}
+              {valueIsInvalid && (
+                <p>Value is not a number</p>
+              )}
             </Tooltip>
           )}
         </Row>
       </div>
     );
   }
 
-  render() {
-    const {
-      minValue, maxValue, id, hoverHint, hoverHintStyle
-    } = this.state;
-
-    const {
-      placeholder, className, label, wide, max, step
-    } = this.props;
-    const doNotInclude = 'wide onMinValueChanged onMaxValueChanged valueMax valueMin ' +
-                         'className type min max step exponent inputOnly ' +
-                         'label noHoverHint noTooltip';
-    const inheritedProps = excludeKeys(this.props, doNotInclude);
-
-    if (this.showTextInput) {
-      return this.renderTextInput();
-    }
-
+  function sliderInput() {
     // Scale values for exponential slider
-    const scaledMinValue = this.valueToSliderPos(minValue);
-    const scaledMaxValue = this.valueToSliderPos(maxValue);
+    const scaledMinValue = valueToSliderPos(minValue);
+    const scaledMaxValue = valueToSliderPos(maxValue);
 
     // HoverHint is in [0, 1]. Scale to full slider range
-    const scaledHoverHintOffset = this.sliderResolution * hoverHint;
-    const tooltipValue = this.valueFromSliderPos(scaledHoverHintOffset);
+    const scaledHoverHintOffset = sliderResolution.current * hoverHint;
+    const tooltipValue = valueFromSliderPos(scaledHoverHintOffset);
 
     return (
       <div
         className={`${styles.inputGroup} ${wide ? styles.wide : ''}`}
-        onDoubleClick={this.enableTextInput}
-        onContextMenu={this.enableTextInput}
-        ref={this.setRef('wrapperRef')}
+        onDoubleClick={enableTextInput}
+        onContextMenu={enableTextInput}
+        ref={wrapperRef}
       >
-        {!this.props.noHoverHint && hoverHint !== null && (
+        {!noHoverHint && hoverHint !== null && (
```

Comment:
```suggestion
        { !noHoverHint && hoverHint !== null && (
```
I know I commented the other way around before, but the aim is to just try to keep it consistent within the same file at least :)